tests/ui/generics/generic-params-nested-fn-scope-error.rs

```diff
@@ -1,9 +1,18 @@
-fn hd<U>(v: Vec<U> ) -> U {
-    fn hd1(w: [U]) -> U { return w[0]; }
-    //~^ ERROR can't use generic parameters from outer item
-    //~| ERROR can't use generic parameters from outer item
+//! Test that generic parameters from an outer function are not accessible
+//! in nested functions. This checks that generic parameter scope is properly
+//! limited to the function where they are declared.
```

Comment:
```suggestion
//! Test that generic parameters from an outer function are not accessible
//! in nested functions.
```

Same thing; better to keep things a bit terse for readability, most of these tests don't really need more than a single sentence describing them